ibm/resource_ibm_private_dns_glb_pool.go

```diff
@@ -0,0 +1,342 @@
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"webhook"}),
+				Description:  "The notification channel",
+			},
+			pdnsGlbPoolRegion: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"us-south", "us-east", "eu-gb", "eu-du", "au-syd", "jp-tok"}),
+				Description:  "Health check region of VSIs",
+			},
+			pdnsGlbPoolSubnet: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Health check subnet IDs of VSIs",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			pdnsGlbPoolCreatedOn: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			pdnsGlbPoolModifiedOn: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceIBMPrivateDNSGLBPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	instanceID := d.Get(pdnsInstanceID).(string)
+	CreatePoolOptions := sess.NewCreatePoolOptions(instanceID)
+
+	poolname := d.Get(pdnsGlbPoolName).(string)
+	CreatePoolOptions.SetName(poolname)
+
+	if description, ok := d.GetOk(pdnsGlbPoolDescription); ok {
+		CreatePoolOptions.SetDescription(description.(string))
+	}
+	if enable, ok := d.GetOk(pdnsGlbPoolEnabled); ok {
+		CreatePoolOptions.SetEnabled(enable.(bool))
+	}
+	if threshold, ok := d.GetOk(pdnsGlbPoolHealthyOriginsThreshold); ok {
+		CreatePoolOptions.SetHealthyOriginsThreshold(int64(threshold.(int)))
+	}
+	if monitor, ok := d.GetOk(pdnsGlbPoolMonitor); ok {
+		CreatePoolOptions.SetMonitor(monitor.(string))
+	}
+	if chanel, ok := d.GetOk(pdnsGlbPoolChannel); ok {
+		CreatePoolOptions.SetNotificationChannel(chanel.(string))
+	}
+	if region, ok := d.GetOk(pdnsGlbPoolRegion); ok {
+		CreatePoolOptions.SetHealthcheckRegion(region.(string))
+	}
+	if _, ok := d.GetOk(pdnsGlbPoolSubnet); ok {
+		CreatePoolOptions.SetHealthcheckSubnets(expandStringList(d.Get(pdnsGlbPoolSubnet).([]interface{})))
+	}
+
+	poolorigins := d.Get(pdnsGlbPoolOrigins).(*schema.Set)
+	CreatePoolOptions.SetOrigins(expandPoolOrigins(poolorigins))
+
+	result, resp, err := sess.CreatePool(CreatePoolOptions)
+	if err != nil {
+		log.Printf("create global load balancer pool failed %s", resp)
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", instanceID, *result.ID))
+	d.Set(pdnsGlbPoolID, *result.ID)
+
+	return resourceIBMPrivateDNSGLBPoolRead(d, meta)
+}
+
+func resourceIBMPrivateDNSGLBPoolRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+	idset := strings.Split(d.Id(), "/")
+
+	getPoolOptions := sess.NewGetPoolOptions(idset[0], idset[1])
+	presponse, resp, err := sess.GetPool(getPoolOptions)
+	if err != nil {
+		return fmt.Errorf("Error fetching pdns GLB Pool:%s\n%s", err, resp)
+	}
+
```

Comment:
Please set all the required attributes.. seems like `pdnsInstanceID` is missing in read method